packages/edit-post/src/components/layout/index.js

```diff
@@ -102,6 +102,7 @@ function Layout() {
 					content={
 						<>
 							<EditorNotices />
+							<Popover.Slot name="block-toolbar" />
```

Comment:
I understand what this is for (the different slot). DOM position and CSS styling but I'm still wondering if we can avoid it. Why can't we  have something that work in a generic way and have the tabbing algorithm be more explicit about moving to toolbar and inspector. For example something like a higher-order component (or a hook) with to wrap any component and that take to optionis to tell it where to go with we tab after the current wrapper and before.

Reply:
Right now the logic is kind of scattered in multiple places (FocusCapture and block onKeyDown handler), thinking we can make this more generic.

Reply:
I can't answer with any authority, but I believe it may solve a z-index stacking order issue by being where it is — i.e. z-index only works in context of the same parent node.

Reply:
@youknowriad The DOM position is important for accessibility and tabbing. The toolbar should be _before_ the content so you can tab backwards to to it. And yes, it also fixes the z-index issue.

Reply:
I get it but honestly, i don't think it's something we'd able to guarantee for all popovers unless we have a Slot per popover which is too much. 

Anyway, that's not my main concern though. I explained my concern and proposal here https://github.com/WordPress/gutenberg/pull/18779#pullrequestreview-334459206

Reply:
I think it's fine for many popovers as we don't need users to be able to tab into them from certain points. However, if some popovers needs a specific position, they can create their own slot.

Reply:
It should be noted that the ideal accessibility situation would be that the toolbar comes after the block, both visually and in tab order.

Reply:
See #3976.

Reply:
Sure, if the visual position changes, we can move it after the content, but right now the toolbar is visually before the block.